### DISCUSSION TITLE
esp32 board with efficient regulator

### DISCUSSION THREAD

**Discussion by kjm1102** [micropython/micropython#12179] [Hardware & Peripherals]
My fav esp32 boards are ones that come with upython already installed. At a pinch I can use Thonny to install the latest upython. The problem is most espp32 boards don't have an efficient regulator for battery applications.

Does anyone have an experience with upython on https://www.waveshare.com/wiki/ESP32-S2-Pico? It's got a nice reg but the rigmarole described to get upython on it seems daunting, I've never had any success with VS code or any of the espressif toolchains.

**Comment by DavesCodeMusings** [comment]
No experience with the board you mentioned, but i do have [a couple of VS Code extensions](https://marketplace.visualstudio.com/publishers/DavesCodeMusings) that might help you with the VS Code aspect you mentioned. The ESPTool one takes care of the firmware flashing. The MPRemote one lets you use MicroPython's [MPRemote](https://docs.micropython.org/en/latest/reference/mpremote.html) commands in a point and click fashion.

**Reply by kjm1102** [reply]
Are you the devil? I've wasted sooo much time trying to get VS code to work, earnestly following various online guides. It never manages to find the esp32 board then says 'no problems have been detected in the workspace' just to taunt me. I hate it with a passion.

**Reply by DavesCodeMusings** [reply]
I am not the prince of darkness, nor do I play him on TV. In all seriousness though, the extensions don't do anything spectacular. They're simply wrappers around the Python modules that do the work. I wrote the extensions because I'd rather point and click than type command-line args.

**Reply by kjm1102** [reply]
Well I installed your MPRemote because point & click would be nice but still no esp32 to be found &, of course, no problems in the workspace as ever.  

Normally I get nothing but dumb insolence from VS code but this time some output in the problem free workspace...Error running command mpremote.devs: command 'mpremote.devs' not found. This is likely caused by the extension that contributes mpremote.devs.

**Reply by DavesCodeMusings** [reply]
If you're interested in working through it, you can post an issue on my github page: https://github.com/DavesCodeMusings/mpremote-vscode/issues

**Reply by kjm1102** [reply]
At your leisure squire, seems to be a terminal issue?

**Comment by peterhinch** [comment]
@kjm1102 Have you seen the [FeatherS3](https://esp32s3.com/feathers3.html)? This has two efficient 3.3V regulators, one of which can be switched off under program control. This enables external devices to be powered down before entering deepsleep. The board has a deepsleep current below 50μA and is ideally suited for battery applications.

**Reply by kjm1102** [reply]
Nice hardware shame about the circuit python

**Reply by scruss** [reply]
it's got releases and nightlies in MicroPython: [FeatherS3](https://micropython.org/download/feathers3/)
